libs/framework/src/bundle_context.c

```diff
@@ -1058,11 +1082,10 @@ bool celix_bundleContext_useServiceWithId(
         const char *serviceName,
         void *callbackHandle,
         void (*use)(void *handle, void *svc)) {
-    celix_service_use_options_t opts = CELIX_EMPTY_SERVICE_USE_OPTIONS;
-
-    char filter[64];
-    snprintf(filter, 64, "(%s=%li)", CELIX_FRAMEWORK_SERVICE_ID, serviceId);
+    char filter[32]; //20 is max long length
+    (void)snprintf(filter, sizeof(filter), "(%s=%li)", CELIX_FRAMEWORK_SERVICE_ID, serviceId);
```

Comment:
A long can take value `INT64_MIN`, which need 20 bytes(including `-`). The total length of the buffer needs to be 1 (`(`) + 10 (CELIX_FRAMEWORK_SERVICE_ID) + 1 (`=`) + 20 (`%li`) + 1 (`)`) + 1 ('\0') = 34 bytes. This modification will lead to gcc warning complaining string truncation.